ocis-pkg/runner/types.go

```diff
@@ -0,0 +1,35 @@
+package runner
+
+// Runable represent a task that can be executed by the Runner.
+// It expected to be a long running task with an indefinite execution time,
+// so it's suitable for servers or services.
+// The task can eventually return an error, or nil if the execution finishes
+// without errors
+type Runable func() error
+
+// Stopper represent a function that will stop the Runable.
+// The stopper acts as a notification to the runable to know that the task
+// needs to be finished now.
+//
+// The stopper won't need to crash the runable or force the runable to stop,
+// instead, it will let the runable to know it has to stop and let it finish.
+// This means that the runable might still run for a while.
+//
+// It's recommended the stopper to run asynchronously. This means that the
+// stopper might need to spawn a goroutine. The intention is avoid blocking
+// the running thread.
+//
+// Usually, the stoppers are the servers's `Shutdown()` or `Close()` methods,
+// that will cause the server to start its shutdown procedure. As said, there
+// is no need to force the shutdown, so graceful shutdowns are preferred if
+// they're available
+type Stopper func()
```

Comment:
Couldn't this return an `error` maybe to signal the runner we won't be able to stop the process?

Reply:
Right now, there are no assumptions on how the stopper needs to behave other than it must notify the task to finish.
The stopper can spawn a new goroutine if the notification can take a long time for whatever reason. This way, it won't block the notification of other services. In this case, returning anything would make the code more complex because we'll need to wait for the response.
Luckily, most of the `Stop` functions don't take long, so we don't need to spawn a goroutine, but it's an option I'd rather keep.

In addition, if the stopper can't stop the process, what can we do? I don't think we can do anything more than whatever the stopper could have tried. I mean, the stopper could also be like
```
func() {
  for i:=0; i<5; i++ {
    if err := server.Stop(); err == nil {
      break
    }
  }
}
```
Any other alternative way to stop the task requires task's knowledge, and it might be a good idea either set it up outside or within the stopper.

Reply:
We could always just `os.Exit` or `panic` to bring all tasks down. I think we need some functionality like this. As mentioned above we must avoid that this loops forever at any cost!

Reply:
true, goroutines run forever when they dont finish themselves. But they won't survive the main process dying. So when you do `os.Exit` all goroutines of this program will stop.